File: classes/006/projects/python/calculator.py
# For command, the first statement is the initialization,
# here we declare a variable its type is defined by the context, in our example
# numeric, this variable will be our state control.
# The second statement is the condition, that will be test at
# the begining of the iteration, is a in (0,1,2,3,4,5,6,7,8,9)
# python will add 1 every time the iteration ends, so a starts at 0,
# and every end of iteration it adds one, so for the first iteration
# a become 1, and so on, then the condition is evaluated again,
# a == 1 in (0,1,2,3,4,5,6,7,8,9), then execute, until a == 10,
# then a in (0,1,2,3,4,5,6,7,8,9) will evaluate to false, so
# our code is no longer executed and the program continues to
# the line after the end of the for command.
for a in range(0,10):
    first = int(input("Enter the first positive integer : "))
    second = int(input("Enter the second positive integer : "))
    operation = input("Which operation ( + | - | * | / ) :")

    if (operation == '+'):
        print("The sum is : " + str(first + second))
    elif (operation == '-'):
        print("The subtraction is : " + str(first - second))
    elif (operation == '*'):
        print("The multiplication is : " + str(first * second))
    elif (operation == '/'):
        if (second == 0):
            print("We cant divide by zero")
        else:
            print("The division is : " + str(first / second))
    else:
        print("Operation not supported ")
# Codes continue from here after the condition is not true anymore

File: classes/006/projects/c++/calculator.cpp
#include <iostream>
#include <string>

int main() {
  // For command, the first statement is the initialization,
  // here we declare a variable of type int starting at zero
  // this variable will be our state control.
  // The second statement is the condition, that will be test at
  // the begining of the iteration, so a < 10 reads as, a lower than 10
  // the third one is the state change statement, we are adding 1 to our
  // control variable, a++ is the same as a + 1, this will be called at
  // the end of each iteration and before the condition check
  // so for our first iteration, a == 0, a is lower than 10, so execute,
  // on the next iteration, we will call a++, changing a == 1,
  // which a still lower than 10, so execute, this will be performed until
  // a++ results in 10, then, a == 10, which makes this condition false, a < 10,
  // well, a is no longer lower than 10, so our code will not execute, and the
  // program execution will continue from the line after the close bracket of our
  // for command.
  for (int a = 0; a < 10; a++) {
    int first  = 0;
    int second = 0;
    std::string operation = "";

    std::cout << "Enter the first positive integer : ";
    std::cin >> first;
    std::cout << "Enter the second positive integer : ";
    std::cin >> second;
    std::cout << "Which operation ( + | - | * | / ) :" ;
    std::cin >> operation;

    if (operation == "+") {
      std::cout << "The sum is : " << first + second << std::endl;
    } else if (operation == "-") {
      std::cout << "The subtraction is : " << first - second << std::endl;
    } else if (operation == "*") {
      std::cout << "The multiplication is : " << first * second << std::endl;
    } else if (operation == "/") {
      if (second == 0) {
        std::cout << "We cant divide by zero" << std::endl;
      } else {
        std::cout << "The division is : " << first / second << std::endl;
      }
    } else {
      std::cout << "Operation not supported" << std::endl;
    }
  } // Closing our for loop block,
  // after the condition evaluates to false we continue from here

  return 0;
}
